paddle/phi/kernels/kps/reduce_sum_kernel.cu

```diff
@@ -13,11 +13,60 @@
 // limitations under the License.
 
 #include "paddle/phi/kernels/reduce_sum_kernel.h"
+#include <climits>
+#include "paddle/phi/core/enforce.h"
 #include "paddle/phi/core/kernel_registry.h"
+#include "paddle/phi/kernels/funcs/eigen/common.h"
 #include "paddle/phi/kernels/gpu/reduce.h"
 
 namespace phi {
 
+template <typename T,
+          int EigenDimSize = 5,
+          int ReducedDimSize = 1,
+          bool ReduceAll = false>
+void ReduceSumEigen(const KPDevice& dev_ctx,
+                    const DenseTensor& x,
+                    bool reduce_all,
+                    const std::vector<int64_t>& dims,
+                    DataType out_dtype,
+                    DenseTensor* out,
+                    std::vector<int>* reduce_dims) {
+  // Resize Input Tensor
+  auto new_x = x;
+  int added_dims = EigenDimSize - x.dims().size();
+  std::vector<int64_t> new_dim(added_dims, 1);
+  for (int i = 0; i < x.dims().size(); i++) {
+    new_dim.push_back(x.dims().at(i));
+  }
+  new_x.Resize(phi::make_ddim(new_dim));
+  auto eigen_x_tensor = EigenTensor<T, EigenDimSize>::From(x);
+
+  // Create Out Tensor
+  dev_ctx.Alloc<T>(out);
+  // Resize Out Tensor
+  std::vector<int64_t> new_reduced_dim(added_dims, 1);
+  for (int i = 0; i < out->dims().size(); i++) {
+    new_reduced_dim.push_back(out->dims().at(i));
+  }
+  out->Resize(phi::make_ddim(new_reduced_dim));
+  constexpr int kReduceOutRank = ReduceAll ? 1 : EigenDimSize - ReducedDimSize;
+  auto eigen_out_tensor = EigenTensor<T, kReduceOutRank>::From(*out);
+  for (int i = 0; i < ReducedDimSize; i++) {
+    (*reduce_dims)[i] += added_dims;
+  }
+  auto eigen_reduce_dim =
+      EigenDim<ReducedDimSize>::From(phi::make_ddim(*reduce_dims));
+  // Caculate
+  eigen_out_tensor.device(*dev_ctx.eigen_device()) =
+      eigen_x_tensor.sum(eigen_reduce_dim);
+  std::vector<int64_t> final_out_dim;
```

Comment:
I prefer to saving the original output dims first like:
```C++
auto origin_out_dims = out->dims();
// some other codes...
out->Resize(origin_out_dims);
```

The code above is more readable.